docs/decisions/005-ballista-extensions.md

```diff
@@ -22,115 +23,156 @@ Related decisions:
 
-### Remote Catalog and UDF Synchronization
+### Catalog and UDF Synchronization Options
 
-1. Require users to manually register tables and UDFs on both scheduler and clients.
-2. Implement catalog and UDF metadata synchronization from scheduler to clients.
-3. Use a shared external catalog service (e.g., Hive Metastore, AWS Glue).
-4. External catalog service: Hive Metastore or AWS Glue would provide catalog synchronization without custom code. However, this would require a custom client build.
+1. **Manual registration**: Require users to manually register tables and UDFs on both scheduler and clients.
+2. **Scheduler-to-client sync**: Implement catalog and UDF metadata synchronization from scheduler to clients, with stub providers for remote tables.
+3. **External catalog service**: Use a shared catalog (e.g., Hive Metastore, AWS Glue) for synchronization. Requires additional infrastructure and custom client build.
+4. **SQL-only interface**: Treat scheduler as the only query planner; clients submit SQL strings only. *(Initial implementation; architecture must support future client-side planning)*
 
 ### Cluster Security
 
-1. Rely on external infrastructure (e.g., service mesh, VPN) for cluster security.
-2. Implement TLS and authentication as optional Ballista configuration.
-3. Use middleware/interceptors for custom security requirements.
-4. Documentation / recommendation to use a service mesh (e.g. Istio) and defer control plane security to the deployer.
+1. **External infrastructure**: Rely on service mesh (Istio), VPN, or network policies for cluster security. Defers security to deployer.
+2. **TLS with API key auth**: Implement TLS and API key authentication as optional Ballista configuration via gRPC interceptors.
+3. **mTLS (mutual TLS)**: Implement mutual TLS for cluster components. Stronger security model but more complex certificate management.
+4. **Middleware/interceptors**: Use gRPC interceptors for custom security requirements without modifying Ballista core.
 
 ### Executor Readiness
 
-1. Poll executor endpoints until they respond successfully.
-2. Add explicit readiness signaling through a channel or callback.
-3. Use existing gRPC health check protocols.
+1. **Scheduler-side polling**: Scheduler polls executor health endpoints until they respond. Compatible with HA scheduler designs.
+2. **Executor-to-scheduler signal**: Executors signal readiness through a channel or callback after first poll loop. Simpler but limits HA scheduler designs.
+3. **gRPC health check protocol**: Use existing gRPC health check protocols (standard, well-supported).
+4. **Kubernetes readiness probes**: Rely on K8s orchestration for readiness; Spice runtime doesn't manage executor lifecycle directly.
```

Comment:
This isn't a valid option in this context - the schedulers need to know which executors are alive so they don't assign work to dead executor nodes.